crest/Assets/Crest/Crest/Scripts/Underwater/UnderwaterRenderer.cs

```diff
@@ -0,0 +1,165 @@
+        public bool meniscus = true;
+
+
+        [Header("Advanced")]
+
+        [Tooltip("Copying params each frame ensures underwater appearance stays consistent with ocean material params. Has a small overhead so should be disabled if not needed.")]
+        public bool copyOceanMaterialParamsEachFrame = true;
+
+        [Predicated("useHorizonSafetyMarginMultiplier", inverted: true), Range(0f, 1f)]
+        [Tooltip("Adjusts the far plane for horizon line calculation. Helps with horizon line issue. (Experimental)")]
+        public float farPlaneMultiplier = 0.68f;
+
+        [Tooltip("Use the old horizon safety margin multiplier to fix horizon line issues instead of the new experimental far plane multiplier.")]
+        public bool useHorizonSafetyMarginMultiplier = false;
+
+        // A magic number found after a small-amount of iteration that is used to deal with horizon-line floating-point
+        // issues. It allows us to give it a small *nudge* in the right direction based on whether the camera is above
+        // or below the horizon line itself already.
+        [Predicated("useHorizonSafetyMarginMultiplier"), Range(0f, 1f)]
+        [Tooltip("A safety margin multiplier to adjust horizon line based on camera position to avoid minor artifacts caused by floating point precision issues, the default value has been chosen based on careful experimentation.")]
+        public float horizonSafetyMarginMultiplier = 0.01f;
+
+        [Predicated("useHorizonSafetyMarginMultiplier"), DecoratedField]
+        [Tooltip("Dynamic resolution can cause the horizon gap issue to widen. Scales the safety margin multiplier to compensate.")]
+        public bool scaleSafetyMarginWithDynamicResolution = true;
+
+        [Space(10)]
+        public DebugFields debug = new DebugFields();
+        [System.Serializable]
+        public class DebugFields
+        {
+            public bool viewOceanMask = false;
+            public bool disableOceanMask = false;
+            public bool disableHeightAboveWaterOptimization = false;
+        }
+
+        Camera _camera;
+        bool _firstRender = true;
+        static int _xrPassIndex = -1;
+
+        // Use instance to denote whether this is active or not. Only one camera is supported.
+        public static UnderwaterRenderer Instance { get; private set; }
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+        static void InitStatics()
+        {
+            Instance = null;
+            _xrPassIndex = -1;
+        }
+
+        void OnEnable()
+        {
+            if (_camera == null)
+            {
+                _camera = GetComponent<Camera>();
+            }
+
+            if (_cameraFrustumPlanes == null)
+            {
+                _cameraFrustumPlanes = GeometryUtility.CalculateFrustumPlanes(_camera);
+            }
+
+            Enable();
+            Instance = this;
+        }
+
+        void OnDisable()
+        {
+            Disable();
```

Comment:
Here we would wrap this and check which RP is active and call the appropriate Disable.